engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/ClassLevelMetaData.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.metadata.aggregated;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.validation.ElementKind;
+
+import org.hibernate.validator.internal.engine.valueextraction.ValueExtractorManager;
+import org.hibernate.validator.internal.metadata.core.ConstraintHelper;
+import org.hibernate.validator.internal.metadata.core.MetaConstraint;
+import org.hibernate.validator.internal.metadata.descriptor.ClassLevelDescriptorImpl;
+import org.hibernate.validator.internal.metadata.raw.ConstrainedElement;
+import org.hibernate.validator.internal.metadata.raw.ConstrainedType;
+import org.hibernate.validator.internal.util.TypeResolutionHelper;
+
+/**
+ * Represents the constraint related meta data for a type i.e. class-level
+ * constraints.
+ *
+ * @author Gunnar Morling
+ * @author Guillaume Smet
+ * @author Marko Bekhta
+ */
+public class ClassLevelMetaData extends AbstractConstraintMetaData {
```

Comment:
+1 for this change.

I would call it `ClassMetaData` though, considering we don't have the `Level` part for the others.